code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -160,22 +160,27 @@ nobiliumsuppression = INFINITY
 	cached_results["fire"] = 0
 	var/turf/open/location = isturf(holder) ? holder : null
 	var/burned_fuel = 0
+
 	if(cached_gases[/datum/gas/oxygen][MOLES] < cached_gases[/datum/gas/tritium][MOLES] || MINIMUM_TRIT_OXYBURN_ENERGY > air.thermal_energy())
 		burned_fuel = cached_gases[/datum/gas/oxygen][MOLES] / TRITIUM_BURN_OXY_FACTOR
 		cached_gases[/datum/gas/tritium][MOLES] -= burned_fuel
+
+		ASSERT_GAS(/datum/gas/water_vapor, air) //oxygen+more-or-less hydrogen=H2O
+		cached_gases[/datum/gas/water_vapor][MOLES] += burned_fuel / TRITIUM_BURN_OXY_FACTOR
+		
 	else
-		burned_fuel = cached_gases[/datum/gas/tritium][MOLES] * TRITIUM_BURN_TRIT_FACTOR
-		cached_gases[/datum/gas/tritium][MOLES] -= cached_gases[/datum/gas/tritium][MOLES] / TRITIUM_BURN_TRIT_FACTOR
-		cached_gases[/datum/gas/oxygen][MOLES] -= cached_gases[/datum/gas/tritium][MOLES]
+		burned_fuel = cached_gases[/datum/gas/tritium][MOLES]
+
+		cached_gases[/datum/gas/tritium][MOLES] -= burned_fuel / TRITIUM_BURN_TRIT_FACTOR
+		cached_gases[/datum/gas/oxygen][MOLES] -= burned_fuel
```

Comment:
Could you elaborate on why you thing `TRITIUM_BURN_TRIT_FACTOR` was used improperly here before?

Reply:
it doesnt make sense that burned_fuel is multiplied by 10. I just moved it around for the sake of sanity and to better reflect the (supposed) meaning of the variable

Reply:
By the looks of things burned_fuel is used for h2o/rad output and dealing with hotspots, whereas the the burning of trit/o2 is based purely on the amount of trit.

Is your goal to change the behavior of the reaction? or to clean up the code. Cause what you're doing not for sure changes behavior, not least of which because trit fires will burn out more easily in this special case.

Reply:
I initially wanted to just make a surgical change to the o2 consumption rate and also fix the burned_fuel snowflake bit, but i forgot to account for the rad output etc. It seems like this will indeed make it quite the behavior change. I'll try and account for everything and keep the behavior on the next commits.